src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IUnaryOperatorExpression.cs

```diff
@@ -0,0 +1,2718 @@
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.Int32)
+  IConversionExpression (ConversionKind.CSharp, Implicit) (OperationKind.ConversionExpression, Type: System.Int32)
+    IInvocationExpression ( System.Char A.Method()) (OperationKind.InvocationExpression, Type: System.Char)
+      Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_Method_BitwiseNot_System_Decimal()
+        {
+            string source = @"
+class A
+{
+    System.Decimal Method()
+    {
+        System.Decimal i = default(System.Decimal);
+        return /*<bind>*/~Method()/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  IInvocationExpression ( System.Decimal A.Method()) (OperationKind.InvocationExpression, Type: System.Decimal)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_Method_BitwiseNot_System_Single()
+        {
+            string source = @"
+class A
+{
+    System.Single Method()
+    {
+        System.Single i = default(System.Single);
+        return /*<bind>*/~Method()/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  IInvocationExpression ( System.Single A.Method()) (OperationKind.InvocationExpression, Type: System.Single)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_Method_BitwiseNot_System_Double()
+        {
+            string source = @"
+class A
+{
+    System.Double Method()
+    {
+        System.Double i = default(System.Double);
+        return /*<bind>*/~Method()/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  IInvocationExpression ( System.Double A.Method()) (OperationKind.InvocationExpression, Type: System.Double)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKin
```

Comment:
not sure how to get DynamicTrue and DynamicFalse
http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/IExpression.cs,544